### PR TITLE
Share xml code in the lexer.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv

**Comment by jaredpar** [comment]
The CI failures seem valid here. Looks like the compiler can no longer bootstrap with this change.

**Comment by CyrusNajmabadi** [comment]
@jcouv @gafter  can you ptal?  Thanks!

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs [hunk 1]
```diff
@@ -3156,6 +3153,15 @@ private bool ScanXmlToken(ref TokenInfo info)
             return info.Kind != SyntaxKind.None;
         }
 
+        private TokenInfo ScanXmlTextLiteralNewLineToken(ref TokenInfo info)
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Parser/Lexer.cs]: Why pass by ref and return? I would expect one or the other.

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs [hunk 2]
```diff
@@ -3682,10 +3688,7 @@ private bool ScanXmlAttributeTextToken(ref TokenInfo info)
 
                 case '\r':
                 case '\n':
-                    this.ScanEndOfLine();
-                    info.StringValue = info.Text = TextWindow.GetText(false);
-                    info.Kind = SyntaxKind.XmlTextLiteralNewLineToken;
-                    this.MutateLocation(XmlDocCommentLocation.Exterior);
+                    ScanXmlTextLiteralNewLineToken(ref info);
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Parser/Lexer.cs]: Doesn't look like the return value is used anywhere.  Think we should make it a void method.

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/Lexer.cs]: No good reason.  Fixed.

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Parser/Lexer.cs]: Agred.  Fixed.

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL

**Comment by CyrusNajmabadi** [comment]
@jcouv Feel free to merge when you want.

**Comment by jaredpar** [comment]
@jcouv was too slow 😄

**Comment by CyrusNajmabadi** [comment]
Ain't that the truth :-P

**Pull request merged**